lld/COFF/Driver.cpp

```diff
@@ -2291,6 +2291,12 @@ void LinkerDriver::linkerMain(ArrayRef<const char *> argsArr) {
   if (auto *arg = args.getLastArg(OPT_deffile)) {
     // parseModuleDefs mutates Config object.
     mainSymtab.parseModuleDefs(arg->getValue());
+    if (ctx.hybridSymtab) {
+      // MSVC ignores the /defArm64Native argument on non-ARM64X targets.
+      // It is also ignored if the /def option is not specified.
+      if (auto *arg = args.getLastArg(OPT_defarm64native))
+        ctx.symtab.parseModuleDefs(arg->getValue());
```

Comment:
Side note: The naming of the variables constantly feels unintuitive to me here, where `mainSymtab` is the EC symbol table, while `ctx.symtab` (which similarly feels like the "main" one) is the native one.

We can't probably do much about it, because this is how the linker is supposed to be have, if mimicing link.exe, but we perhaps make it clearer. I think it would be good to have a comment on the definition of `mainSymtab` to really spell out what it is, and why.

Reply:
Yes, which symbol table is considered 'main' or 'primary' depends on the context. In the writer, the main one is `ctx.symtab`, as it’s directly referenced by the PE headers. Here, for lack of a better term, I meant the symbol table to which the main arguments should apply (though not all do yet). I'll add a comment for clarification.